### PR TITLE
Rename rustc guide

### PR DESCRIPTION
**Pull request by spastorino**
This is in preparation for https://github.com/rust-lang/rustc-guide/issues/470
Needs to be merged after we actually rename the guide.

Have used this to rename:

`git grep -l 'rustc_guide' | xargs sed -i 's/rustc_guide/rustc_dev_guide/g'`
`git grep -l 'rustc-guide' | xargs sed -i 's/rustc-guide/rustc-dev-guide/g'`
`git grep -l 'rustc guide' | xargs sed -i 's/rustc guide/rustc dev guide/g'`

### CODE REVIEW

File: .gitmodules
Comment (spastorino) [inline .gitmodules]: Is this change handled gracefully by our build system or do I need to change something else? cc @Mark-Simulacrum

Comment (mark-i-m) [reply .gitmodules]: I think it's fine. But perhaps also git mv to rename the directory.

Comment (spastorino) [reply .gitmodules]: We should update the submodule after the rename. As soon as we do I can fix this PR.

File: CONTRIBUTING.md
Comment (LeSeulArtichaut) [inline CONTRIBUTING.md]: Makes me wonder if we should decide whether to use `rustc-dev-guide` or `rustc dev guide` and stick to one of them, or if it is completely insignificant

File: src/librustc_ast/README.md
Comment (LeSeulArtichaut) [inline src/librustc_ast/README.md]: Maybe add an active link here?

**Review by LeSeulArtichaut — APPROVED** [review]
Looks good to me, haven't spotted any mistake.

File: src/librustc_ast/README.md
Comment (spastorino) [reply src/librustc_ast/README.md]: I think we should, but maybe let's handle that on a separate PR?

File: CONTRIBUTING.md
Comment (spastorino) [reply CONTRIBUTING.md]: well I was just naively translating things, unsure if it's significant but I guess we can discuss on a separate PR.

**Comment by JohnTitor** [comment]
36003a2 includes clippy sudmodule, we should drop it.

**Review by mark-i-m — CHANGES_REQUESTED** [review]
Mostly looks good to me, but it looks like you accidentally updated the clippy submodule.

**Comment by spastorino** [comment]
Seems like the repo was dirty because the command I was running doesn't include submodules. Anyway, fixed thanks for pointing that out.

**Review by mark-i-m — APPROVED** [review]
Looks good now!

**Comment by JohnTitor** [comment]
Actually, we have to rename the repo actually to r+, let's hold off on at this point.
r? @JohnTitor

**Comment by spastorino** [comment]
Yeah, this one is not ready. I thought I was changing rustc-guide :P. We need to first rename the repo.

File: .gitmodules
Comment (Mark-Simulacrum) [reply .gitmodules]: I believe this should be fine, though I'm personally uncertain whether something else needs to change (i.e., our tooling should be fine but I don't know what git'll want).

**Comment by spastorino** [comment]
This should be now, ready to merge

**Comment by JohnTitor** [comment]
r=me on behalf of the WG once CI is happy.

**Comment by JohnTitor** [comment]
Ah, it seems the previous submodule remains: https://github.com/rust-lang/rust/tree/f4445da203ca08659f513a841f13cbfbb22b460b/src/doc

**Comment by spastorino** [comment]
fixed!

**Review by mark-i-m — APPROVED** [review]
LGTM

**Comment by pietroalbini** [comment]
r=me once CI passes.

**Pull request merged**
